controllers/ocirepository_controller.go

```diff
@@ -362,6 +364,39 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 		return sreconcile.ResultEmpty, e
 	}
 
+	// Verify the image
+	if obj.Spec.Verify != nil {
+		provider := obj.Spec.Verify.Provider
+		switch provider {
+		case "cosign":
+			if verified, err := r.verify(ctx, obj, url); err != nil {
+				e := serror.NewGeneric(
+					fmt.Errorf("failed to verify '%s' using provider '%s': %w", url, provider, err),
+					sourcev1.SourceVerifiedFailedReason,
+				)
+				conditions.MarkFalse(obj, sourcev1.SourceVerifiedCondition, e.Reason, e.Err.Error())
+				return sreconcile.ResultEmpty, e
+			} else {
+				if !verified {
+					e := serror.NewGeneric(
+						fmt.Errorf("no matching signatures '%s' using provider '%s': %w", url, provider, err),
+						sourcev1.SourceVerifiedFailedReason,
+					)
+					conditions.MarkFalse(obj, sourcev1.SourceVerifiedCondition, e.Reason, e.Err.Error())
+					return sreconcile.ResultEmpty, nil
+				}
```

Comment:
Maybe we should do something like:
```
conditions.MarkTrue(obj, sourcev1.SourceVerifiedCondition, "Reason", "Message")
```
and the message can be something like `OCI Image with hash %s verified.`

Reply:
that makes a lot of sense, thanks.

```go
conditions.MarkTrue(obj, sourcev1.SourceVerifiedCondition, "OCI Image %s with digest %s verified.", url, imgDigest)
```